pkg/core/resources/apis/mesh/validators.go

```diff
@@ -370,6 +378,31 @@ func ValidateTargetRef(
 	return err
 }
 
+func DoesTargetRefSupportGateway(ref common_api.TargetRef) bool {
```

Comment:
Kind of a nit but maybe this should just be a method on `common_api.TargetRef` called `IncludesGateways` or similar